drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/models/AbstractScesimGridModel.java

```diff
@@ -1055,6 +1070,23 @@ protected void refreshErrorsRow(int rowIndex) {
         });
     }
 
+    /**
+     * Returns the correct <b>DOMElement</b> factory to the given data
+     * @param modelType
+     * @param valueType
+     */
+    public BaseSingletonDOMElementFactory getDOMElementFactory(String className,
+                                                               ScenarioSimulationModel.Type modelType,
+                                                               FactMappingValueType valueType) {
+        boolean isRuleScenario = Objects.equals(ScenarioSimulationModel.Type.RULE, modelType);
+        if (ScenarioSimulationSharedUtils.isCollection(className)) {
+            return collectionEditorSingletonDOMElementFactory;
+        } if (Objects.equals(FactMappingValueType.EXPRESSION, valueType) && isRuleScenario) {
+            return scenarioExpressionCellTextAreaSingletonDOMElementFactory;
+        }
+        return scenarioCellTextAreaSingletonDOMElementFactory;
```

Comment:
Could unexpectedly return null. Shouldn't we initialise it first?

Reply:
@dupliak What you mean with "initialize" it? Are you referring to the factories?

Reply:
I mean that once we will return here a scenarioCellTextAreaSingletonDOMElementFactory set to null what would happen next?

Reply:
It is dupliaka)

Reply:
@dupliaka In case of a null factory, the calling method will assign to its referring column a null factory. I think this will lead to an error (probably a NPE) when you try to edit a value cell. Anyway, I assume all the factories are initialized during the creation of `AbstractScesimGridModel`. If this it's false, then it's correct to thrown an error, imho.